### PR TITLE
[tests] functional tests should call BitcoinTestFramework start/stop node methods

### PR DESCRIPTION
**Pull request by jnewbery**
Second step towards #10082

This PR changes the individual test cases to call the `BitcoinTestFramework.start_node` and `.stop_node` methods rather than the util.py functions. This is in preparation for moving the start/stop node functionality into BitcoinTestFramework (currently the BitcoinTestFramework start/stop node methods are just wrappers for the util.py functions).

The second commit changes the names of the functions in util.py to mark them as private. This is so no test cases are added/modified to call the functions directly before the next step of #10082 is merged.

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Concept ACK. Needs rebase.

**Comment by jnewbery** [comment]
rebased

**Comment by jnewbery** [comment]
rebased. It'd be really helpful to get this merged so we can make some progress on #10082 . This is the only disruptive changeset in that larger PR - the rest of the changes are restricted to the test_framework directory.

This PR touches a lot of files, but it should be a trivial review. It's simply changing calls from `stop_node()` to calls to `self.stop_node()` etc wrapper functions in the first commit. The second commit renames the inner functions so no-one calls them explicitly from the test cases (the inner functions will be removed entirely in a future PR).

@MarcoFalke , @TheBlueMatt , @ryanofsky , @sdaftuar - if I could convince any of you to review, I'd be very grateful!

File: test/functional/abandonconflict.py [hunk 1]
```diff
@@ -73,8 +73,8 @@ def run_test(self):
 
         # Restart the node with a higher min relay fee so the parent tx is no longer in mempool
         # TODO: redo with eviction
-        stop_node(self.nodes[0],0)
-        self.nodes[0]=start_node(0, self.options.tmpdir, ["-minrelaytxfee=0.0001"])
+        self.stop_node(0)
+        self.nodes[0] = self.start_node(0, self.options.tmpdir, ["-minrelaytxfee=0.0001"])
```
Comment (jimmysong) [inline test/functional/abandonconflict.py]: The plan is to change `start_node` so assignment here would be unnecessary in the future?

File: test/functional/blockchain.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
     assert_raises_jsonrpc,
     assert_is_hex_string,
     assert_is_hash_string,
+    connect_nodes_bi,
```
Comment (jimmysong) [inline test/functional/blockchain.py]: Not sure why this was added?

Comment (ryanofsky) [inline test/functional/blockchain.py]: Change seems unrelated, maybe revert.

File: test/functional/fundrawtransaction.py [hunk 3]
```diff
@@ -449,12 +449,12 @@ def run_test(self):
...
+        self.stop_node(0)
+        self.stop_node(2)
+        self.stop_node(3)
         self.nodes.pop(1)
-        stop_node(self.nodes[0], 0)
-        stop_node(self.nodes[1], 2)
-        stop_node(self.nodes[2], 3)
 
-        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir)
+        self.nodes = self.start_nodes(self.num_nodes, self.options.tmpdir)
```
Comment (jimmysong) [inline test/functional/fundrawtransaction.py]: I assume that `self.start_nodes` would also do the assignment in the future?

File: test/functional/net.py [hunk 4]
```diff
@@ -14,8 +14,7 @@
     assert_equal,
     assert_raises_jsonrpc,
     connect_nodes_bi,
-    p2p_port,
-)
+    p2p_port)
```
Comment (jimmysong) [inline test/functional/net.py]: Nit: I personally like comma endings so adding new imports has cleaner diffs.

File: test/functional/fundrawtransaction.py [hunk 5]
```diff
@@ -449,12 +449,12 @@ def run_test(self):
         ############################################################
         # locked wallet test
         self.nodes[1].encryptwallet("test")
+        self.stop_node(0)
```
Comment (ryanofsky) [inline test/functional/fundrawtransaction.py]: Maybe move these up one line so nodes[1].encrypt and pop(1) can be next to each other.

**Review by ryanofsky — COMMENTED** [review]
Slightly tested ACK a433d8a15e165cfb42e44319d608365603f64e0c

**Review by jimmysong — APPROVED** [review]
ACK. testing now.

**Comment by jimmysong** [comment]
Tested, ACK a433d8a15e165cfb42e44319d608365603f64e0c

A couple of really small nits. Excited to keep this moving!

File: test/functional/fundrawtransaction.py (hunk 3 above)
Comment (kallewoof) [inline test/functional/fundrawtransaction.py]: Unless it should be possible to append nodes.

File: test/functional/net.py (hunk 4 above)
Comment (kallewoof) [inline test/functional/net.py]: Agreed

**Review by kallewoof — APPROVED** [review]
utACK a433d8a15e165cfb42e44319d608365603f64e0c

File: test/functional/blockchain.py (hunk 2 above)
Comment (jnewbery) [inline test/functional/blockchain.py]: yes. not sure how that snuck in. Reverted.

File: test/functional/abandonconflict.py (hunk 1 above)
Comment (jnewbery) [inline test/functional/abandonconflict.py]: yes - that's certainly something we can do in a future PR once the BitcoinTestFramework class is responsible for tracking and stop/starting the nodes.

File: test/functional/blockchain.py (hunk 2 above)
Comment (jnewbery) [inline test/functional/blockchain.py]: me neither :flushed:. Removed

File: test/functional/fundrawtransaction.py (hunk 3 above)
Comment (jnewbery) [inline test/functional/fundrawtransaction.py]: certainly something we can do in a future PR.

BitcoinTestFramework could include two functions - `start_nodes()` to start the nodes in `self.nodes`, and `add_nodes()` if you need to append nodes to that list.

File: test/functional/net.py (hunk 4 above)
Comment (jnewbery) [inline test/functional/net.py]: Sure. Reverted.

**Comment by jnewbery** [comment]
Thanks @ryanofsky @jimmysong @kallewoof . Nits fixed in fixup commit. Will squash when this is ready for merge.

**Comment by MarcoFalke** [comment]
utACK 53f6775
